builtin/k8s/platform.go

```diff
@@ -600,7 +600,7 @@ func (p *Platform) Deploy(
 	sg := ui.StepGroup()
 	defer sg.Wait()
 
-	if p.config.ServicePort == 0 {
+	if p.config.ServicePort == 0 && p.config.Ports == nil {
 		p.config.ServicePort = 3000
 	}
```

Comment:
I think this line might not be required at all, it looks like it showed up during the k8s resource manager conversion. ​I thought this if-block was covering configuring the single and multi-port case:

https://github.com/hashicorp/waypoint/blob/bb6e0060ac0d2f5a8726c3b8a103e8991531e989/builtin/k8s/platform.go#L167-L180


If removing this doesn't fix it, we should try to fix up the port conversion instead over in `resourceDeploymentCreate` and remove this here.

Reply:
@briancain You're absolutely right: I've removed that code, and things work without the error.